Bug 1723798: bootkube: run all podman commands in the host network.

squeed opened the pull request:
We had an issue ([bz 1723798](https://bugzilla.redhat.com/show_bug.cgi?id=1723798)) where bootstrap failed because the default podman IP range conflicted with the user's internal range. Since we don't rely on podman for network isolation, the easiest way to fix this is just to run all bootstrapping pods in the host network.

squeed commented:
/retest

squeed commented:
/retest

cgwalters commented on data/data/bootstrap/files/usr/local/bin/bootkube.sh.template:
```diff
@@ -16,23 +16,23 @@ if ! release=$( podman inspect {{.ReleaseImage}} -f '{{"{{"}} index .RepoDigests
-MACHINE_CONFIG_ETCD_IMAGE=$(podman run --quiet --rm ${release} image etcd)
-MACHINE_CONFIG_KUBE_CLIENT_AGENT_IMAGE=$(podman run --quiet --rm ${release} image kube-client-agent)
-MACHINE_CONFIG_INFRA_IMAGE=$(podman run --quiet --rm ${release} image pod)
+MACHINE_CONFIG_OPERATOR_IMAGE=$(podman run --quiet --rm --net=host ${release} image machine-config-operator)
```
Nit; some of these, in fact most could be `--net=none` (not that it matters really but being able to disable networking is a powerful container feature in general and we should keep it in mind).

Also may be worth a prep commit which abstracts a `bootkube-podman-run` helper or so.

squeed commented:
/retest
This might need to be rebased. If so, I'll tweak the commands.

squeed commented:
@cgwalters updated it a bit - PTAL

cgwalters commented:
LGTM!

abhinavdahiya commented:
/assign @wking

wking commented on data/data/bootstrap/files/usr/local/bin/bootkube.sh.template:
```diff
@@ -16,23 +16,26 @@ if ! release=$( podman inspect {{.ReleaseImage}} -f '{{"{{"}} index .RepoDigests
-MACHINE_CONFIG_KUBE_CLIENT_AGENT_IMAGE=$(podman run --quiet --rm ${release} image kube-client-agent)
-MACHINE_CONFIG_INFRA_IMAGE=$(podman run --quiet --rm ${release} image pod)
+bootkube_podman_run="podman run --quiet --net=host"
+image_for="podman run --quiet --rm --net=none ${release} image"
```
ShellCheck doesn't seem to have a problem with your approach, but I'd be more comfortable with shell functions:

```sh
bootkube_podman_run() {
  podman run --quiet --net=host "${@}"
}

image_for() {
  RELEASE="${1}"
  podman run --quiet --rm --net=none "${RELEASE}" image
}

MACHINE_CONFIG_OPERATOR_IMAGE=$(image_for machine-config-operator)
...
  bootkube_podman_run \
    --volume "$PWD:/assets:z" \
...
```

squeed replied:
hah, that's because we don't shellcheck bootkube.sh.

squeed commented:
Rebased. Removed the shellcheck comments, since they're false security.

I ran the file through a go template evaluator and shellchecked it. It passed. Probably something to add to CI, but not in this PR :-).

wking commented on data/data/bootstrap/files/usr/local/bin/bootkube.sh.template:
```diff
@@ -16,28 +16,38 @@ if ! release=$( podman inspect {{.ReleaseImage}} --format '{{"{{"}} index .RepoD
 
-KUBE_ETCD_SIGNER_SERVER_IMAGE=$(podman run --quiet --rm ${release} image kube-etcd-signer-server)
+image_for() {
+    podman run --quiet --rm --net=none ${release} image "${1}"
```
Quote `"${release}"`

squeed replied:
done.

wking commented:
/lgtm
